go/arrow/array/encoded.go

```diff
@@ -68,6 +69,65 @@ func (r *RunEndEncoded) Release() {
 	r.ends.Release()
 }
 
+// LogicalValuesArray returns an array holding the values of each
+// run, only over the range of run values inside the logical offset/length
+// range of the parent array.
+//
+// The return from this needs to be Released
+func (r *RunEndEncoded) LogicalValuesArray() arrow.Array {
+	physOffset := r.GetPhysicalOffset()
+	physLength := r.GetPhysicalLength()
+	data := NewSliceData(r.data.Children()[1], int64(physOffset), int64(physOffset+physLength))
+	defer data.Release()
+	return MakeFromData(data)
+}
+
+// LogicalRunEndsArray returns an array holding the logical indexes
+// of each run end, only over the range of run end values relative
+// to the logical offset/length range of the parent array.
+//
+// For arrays with an offset, this is not a slice of the existing
+// internal run ends array.
```

Comment:
Would be nice to show an example here as well (preferably the same as in `LogicalValuesArray`).

Reply:
added